### PR TITLE
Spec: required request bodies

### LINKED ISSUE #10004: OpenApi requestBody: some are optional but should be required
### Apache Iceberg version

1.5.0 (latest release)

### Query engine

None

### Please describe the bug 🐞

`requestBody` is [optional by default](https://swagger.io/docs/specification/describing-request-body) and needs `required: true` to override that. It seems the following should be required but are currently optional.
1. [createNamespace](https://github.com/apache/iceberg/blob/f425dc7401b268f92778c3c9b2c6abfc7fb661df/open-api/rest-catalog-open-api.yaml#L250). How else to know the namespace to create?
1. [updateProperties](https://github.com/apache/iceberg/blob/f425dc7401b268f92778c3c9b2c6abfc7fb661df/open-api/rest-catalog-open-api.yaml#L398). How else to know what properties to update?
1. [createTable](https://github.com/apache/iceberg/blob/f425dc7401b268f92778c3c9b2c6abfc7fb661df/open-api/rest-catalog-open-api.yaml#L501). How else to know the table to create?
1. [updateTable](https://github.com/apache/iceberg/blob/f425dc7401b268f92778c3c9b2c6abfc7fb661df/open-api/rest-catalog-open-api.yaml#L682). How else to know what to commit?
1. [registerTable](https://github.com/apache/iceberg/blob/f425dc7401b268f92778c3c9b2c6abfc7fb661df/open-api/rest-catalog-open-api.yaml#L553). How else to know the table to register?
1. [createView](https://github.com/apache/iceberg/blob/f425dc7401b268f92778c3c9b2c6abfc7fb661df/open-api/rest-catalog-open-api.yaml#L1113). How else to know the view to create?
1. [replaceView](https://github.com/apache/iceberg/blob/f425dc7401b268f92778c3c9b2c6abfc7fb661df/open-api/rest-catalog-open-api.yaml#L1211). How else to know what to commit?

Tools that generate code from the spec dutifully make these optional, forcing a developer to handle this confusing and unnecessary case. Can these be marked as required?

### PR DESCRIPTION
**Pull request by westse**
Closes #10004

### CODE REVIEW

File: open-api/rest-catalog-open-api.yaml [hunk 1]
```diff
@@ -248,6 +248,7 @@ paths:
         The server might also add properties, such as `last_modified_time` etc.
       operationId: createNamespace
       requestBody:
+        required: true
```
Comment (nastra) [inline open-api/rest-catalog-open-api.yaml:251]: can you also please add the same for `getToken`?

Comment (westse) [reply open-api/rest-catalog-open-api.yaml:252]: Done

**Review by nastra — APPROVED** [review]

**Review by amogh-jahagirdar — APPROVED** [review]
Good find @westse ! the bodies should be required. I diffed the cases where already specified required: true, and what we were missing and this covers the remainder!

**Comment by amogh-jahagirdar** [comment]
I'll go ahead and merge, thanks @westse for fixing this and @nastra for reviewing.

**Pull request merged**